daemon/snapshotter/mount.go

```diff
@@ -14,7 +14,7 @@ import (
 const mountsDir = "rootfs"
 
 // List of known filesystems that can't be re-mounted or have shared layers
-var refCountedFileSystems = []string{"overlayfs", "zfs", "fuse-overlayfs"}
+var refCountedFileSystems = []string{"fuse-overlayfs", "overlayfs", "stargz", "zfs"}
```

Comment:
Perhaps we need a tracking ticket for this hard-coded list;

- is this information that the snapshotter itself could provide?
- If we _must_ hard-code, do we need to add more  snapshotters to this list?

Reply:
I also wonder if we should just go the route of only using the refcounted method, myself.

Reply:
Good one; perhaps yes. Can you create a tracking ticket for further discussion? (or a draft PR that removes it, which could also help with the discussion)

Reply:
Going with the refcount in all cases seems good to me too.

Also, If re-mount not being supported is a norm, rather than exception, we could go with the reverse - disabling ref counting only for snapshotters that are known to support re-mount and have been tested.

Reply:
Yeah, these are the kind of things I'd love to _either_ see centralised (in containerd library code), or to have an API to request the information from the snapshotter (or some other way to feature-detect). Especially because the list of possible snapshotters is "unlimited", so we can't _really_ make any assumptions beyond the ones we happen to know of.

but perhaps @kzys and @AkihiroSuda have thoughts on that as well

Reply:
Yes, this information should be exposed to the plugin API

Reply:
- opened https://github.com/moby/moby/issues/46842